src/parse.jl

```diff
@@ -0,0 +1,254 @@
+"""
+    count_locations_and_transitions(root_sxmodel)
+
+Returns the number of locations and transitions for each component.
+
+### Input
+
+- `root_sxmodel` -- the root element of a SX file
+
+### Output
+
+Two vectors of integers `(lcount, tcount)`, where the i-th entry of `lcount` and
+`tcount` are number of locations and transitions, respectively, of the i-th component.
+"""
+function count_locations_and_transitions(root_sxmodel)
+    lcount = Vector{Int}() # num locations for each component
+    tcount = Vector{Int}() # num transitions for each component
+    id_component = 0 # index of the current component
+
+    # count the number of locations for each component
+    for component in eachelement(root_sxmodel)
+        if nodename(component) == "component"
+            id_component += 1
+            push!(lcount, 0)
+            push!(tcount, 0)
+            for field in eachelement(component)
+                if nodename(field) == "location"
+                    lcount[id_component] += 1
+                elseif nodename(field) == "transition"
+                    tcount[id_component] += 1
+                end
+            end
+        end
+    end
+    return (lcount, tcount)
+end
+
+"""
+    parse_sxmath(s)
+
+Returns the list of expressions corresponding to a given SX string.
+
+### Input
+
+- `s` - string
+
+### Output
+
+Vector of expressions, equations or inequalities.
+
+### Examples
+
+```jldoctest
+julia> import SX.parse_sxmath
+
+julia> parse_sxmath("x >= 0")
+1-element Array{Expr,1}:
+ :(x >= 0)
+
+julia> parse_sxmath("x' == x & v' == -0.75*v")
+2-element Array{Expr,1}:
+ :(x' = x)
+ :(v' = -0.75v)
+
+julia> parse_sxmath("x == 0 & v <= 0")
+2-element Array{Expr,1}:
+ :(x = 0)
+ :(v <= 0)
+```
+
+### Algorithm
+
+Consists of the following steps (in the given order):
+
+- split the string with the `&` key
```

Comment:
Do you want to allow parentheses? Since the formulas are purely conjunctive, they could just be ignored.

Reply:
good point. as a by-product of `parse`, parentheses are ignored:

```julia
julia> parse_sxmath("(x == 0) & ((v <= 0))")
2-element Array{Expr,1}:
 :(x = 0)
 :(v <= 0)
```

Reply:
note that & is the same as && for SpaceEx

Reply:
>  note that & is the same as && for SpaceEx

Thanks. I've updated the function `parse_sxmath` accordingly.